tools/wake-format/emitter.cpp

```diff
@@ -39,123 +59,75 @@ wcl::rope Emitter::newline(ctx_t ctx) {
 wcl::rope Emitter::space(uint8_t count) {
   wcl::rope_builder builder;
   for (uint8_t i = 0; i < count; i++) {
-    builder.append("·");
+    builder.append(" ");
   }
   return std::move(builder).build();
 }
 
 wcl::optional<wcl::rope> Emitter::flat(ctx_t ctx, CSTElement node) {
-  return walk_node(ctx.flat(), node);
-}
-
-wcl::rope Emitter::try_flat(ctx_t ctx, CSTElement node) {
-  auto flat_node = flat(ctx, node);
-  if (flat_node && flat_node->column() <= max_column_width) {
-    return *flat_node;
+  // Try to get the flat version of node
+  auto flat_node = walk_node(ctx.flat(), node);
+  if (!flat_node) {
+    return {};
   }
-  auto full_node = walk_node(ctx, node);
-  assert(full_node);
-  return *full_node;
-}
 
-#define ASSERT_TOKEN(node, token) \
-  assert(!node.empty());          \
-  assert(node.id() == token);
-
-wcl::rope Emitter::layout(CST cst) {
-  ctx_t ctx;
-  return walk_top(ctx, cst.root()).concat(newline(ctx));
-}
-
-wcl::rope Emitter::walk_top(ctx_t ctx, CSTElement node) {
-  wcl::rope_builder builder;
-
-  for (CSTElement child = node.firstChildElement(); !child.empty(); child.nextSiblingElement()) {
-    if (child.isNode()) {
-      builder.append(try_flat(ctx, child));
-      builder.append(newline(ctx));
-    } else {
-      builder.append(walk_token(ctx, child));
-    }
+  if (flat_node->column() > max_column_width) {
```

Comment:
@JakeSiFive  Happy to this condition here if it is still incorrect in the scope of the function changes here

Reply:
You can't get the correct condition in this context. You always have to look at the context that you're appending it into to know if we can you do this or not. You always have to add the size (or column) with the current builder column to know what the new last column will be. What you have now is a context free approximation.

Reply:
actually, I lied, fixing this in the next PR, but the change there is just
```  
if (flat_node->column() + (ctx.nest_level * space_per_indent) > max_column_width) {
  return {};
}
```
which accounts for the missing indentation at the start of the line

Reply:
Ah, hmm, that's not good enough either. I think I'm starting to see why we need the `builder->column()`